test/remote/gateways/remote_checkout_v2_test.rb

```diff
@@ -1186,4 +1191,51 @@ def test_non_truncate_id_for_non_amex_transactions
     assert_equal 31, response.params['reference'].length
     assert_equal 'Visa', response.params['source']['scheme']
   end
+
+  def test_successful_purchase_with_minimun_lvl_2_data
+    response = @gateway.purchase(@amount, @credit_card, @minimun_lvl2_options)
+    assert_success response
+  end
+
+  def test_successful_purchase_with_minimun_lvl_2_data_for_amex
+    amex_options = {
+      currency: 'USD',
+      line_items: [{ item_name: 'Paint', quantity: '1', unit_cost: '1270' }]
+    }
+    @amex_card = credit_card('345678901234564', brand: 'american_express', verification_value: '1000', month: '12', year: Time.now.year)
+
+    response = @gateway.purchase(1500, @amex_card, @minimun_lvl2_options.merge(amex_options))
+    assert_success response
+  end
+
+  def test_successful_purchase_with_minimun_lvl_3_data
+    order_data = {
+      processing: { order_id: '01234' },
```

Comment:
❓ Question:
This one relates to one above => how is order_data[:processing][:order_id] going to be sent in this structure if is a GSF, unless the entire lvl3 data fits in one big GSF

Reply:
In this case only order_id will be passed in the structure, due confusing with the order_id in the transaction, but it can be passed in a different and not nested value like invoice_id.

Is not needed all the values to ActiveMerchant pass in the same structure. But maybe for clarity could be better modify the value name.